### PR TITLE
feat: add tracking params and go straight to app store for iOS for the landing page wallet CTA

### PR DESCRIPTION
**Pull request by natew**
## Description

Adds tracking + go to app store for iOS.

_Linear ticket:_
_Slack thread:_
_Relevant docs:_

### Before
|    Mobile    |   Desktop    |
| ------------ | ------------ |
| paste_before | paste_before |

### After
|    Mobile    |   Desktop   |
| ------------ | ----------- |
| paste_after  | paste_after |

### Reproducing the error

1. 

### QA (ie manual testing)

- [ ] N/A

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

File: src/pages/Landing/index.tsx
Comment (natew) [inline src/pages/Landing/index.tsx:394]: didn't want to add analytics event but will do that now...

Comment (natew) [reply src/pages/Landing/index.tsx:393]: pending here, we can follow on with it https://github.com/Uniswap/analytics-events/pull/47

File: src/utils/openDownloadApp.ts
Comment (adjkant) [inline src/utils/openDownloadApp.ts:23]: Will this just log an element clicked event with no properties if there's no element? Should we only log when there's an element?

File: src/pages/Landing/index.tsx
Comment (adjkant) [reply src/pages/Landing/index.tsx:393]: What is this event capturing vs the other existing one / the options?

File: src/utils/openDownloadApp.ts
Comment (natew) [reply src/utils/openDownloadApp.ts:23]: good question, i defer to @tinaszheng

**Comment by natew** [comment]
@adjkant re:

> What is this event capturing vs the other existing one / the options?

In this case it's just capturing the download vs the wallet landing page, but we need to land analytics tags to pass there instead of `undefined`, but this link doesn't have analytics otherwise so I think it does add a unique event.

File: src/utils/openDownloadApp.ts
Comment (adjkant) [reply src/utils/openDownloadApp.ts:23]: I think it will in this case - once the analytics PR merges we should make sure this doesn't allow an undefined analytics event here / doesn't log any event in that case.

File: src/utils/openDownloadApp.ts
Comment (adjkant) [inline src/utils/openDownloadApp.ts:12]: IMO this is a case where abstracting the functionality confuses more than it helps. Opening the microsite is very different than opening the app store - can we do two minimal param functions for each of these that log their respective event automatically, no param needed beyond the interface element?

**Review by adjkant — CHANGES_REQUESTED** [review]

File: src/utils/openDownloadApp.ts
Comment (natew) [reply src/utils/openDownloadApp.ts:12]: @adjkant i think these are coupled somewhat how they are used though, as we basically use the same link to open one or the other in a few different areas, but in one area we need to customize the params. if i split them out then i'd have to add logic to the call-site in multiple places that is duplicated. open to some third alternative or something though

Comment (natew) [reply src/utils/openDownloadApp.ts]: @adjkant did a pass over this and i think its a bit cleaner.

Comment (natew) [reply src/utils/openDownloadApp.ts]: added the new event as well

File: src/components/Banner/UniswapWalletBanner.tsx
Comment (adjkant) [inline src/components/Banner/UniswapWalletBanner.tsx:137]: any reason not to pass directly as before?

File: src/utils/openDownloadApp.ts
Comment (adjkant) [inline src/utils/openDownloadApp.ts:56]: naming nit: `urlParamString`?

File: src/pages/Landing/index.tsx
Comment (adjkant) [inline src/pages/Landing/index.tsx:391]: Make these strings constants still so it's clear what they represent / the case?

**Review by adjkant — APPROVED** [review]
Looks good on the analytics logging part! Some small nits but otherwise good to go

File: src/utils/openDownloadApp.ts
Comment (adjkant) [reply src/utils/openDownloadApp.ts]: Looks good on this part!

**Comment by adjkant** [comment]
need to fix callsites as well, see typescript issues

File: src/pages/Landing/index.tsx
Comment (adjkant) [reply src/pages/Landing/index.tsx:392]: reopening this one, can we move this object into `src/utils/openDownloadApp.ts` as `landingPageParams`?

**Review by adjkant — APPROVED** [review]
one nit, otherwise looks good!

File: src/utils/openDownloadApp.ts
Comment (zzmp) [inline src/utils/openDownloadApp.ts:34]:
```suggestion
    openWalletMicrosite({ element: options?.element, urlParamString: options?.microSiteParams })
```

File: src/utils/openDownloadApp.ts
Comment (zzmp) [inline src/utils/openDownloadApp.ts:32]:
```suggestion
    openAppStore({ element: options?.element, urlParamString: options?.appStoreParams })
```

File: src/utils/openDownloadApp.ts
Comment (zzmp) [inline src/utils/openDownloadApp.ts:61]: What does `target` do here (and below)?

File: src/pages/Landing/__snapshots__/index.test.tsx.snap
Comment (zzmp) [inline src/pages/Landing/__snapshots__/index.test.tsx.snap:4515]: Can you make sure both the iOS and microsite are tested in a snapshot, by mocking the isIOS hook?

File: src/components/AccountDrawer/DownloadButton.tsx
Comment (zzmp) [inline src/components/AccountDrawer/DownloadButton.tsx:48]: nit: if you remove the comma the linter will allow it as a one-liner

**Review by zzmp — CHANGES_REQUESTED** [review]
I think one of the params is incorrectly passed

File: src/utils/openDownloadApp.ts
Comment (zzmp) [reply src/utils/openDownloadApp.ts:34]: Shouldn't this be `options?.microSiteParams`?

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**
